### DISCUSSION TITLE
Run nextcloud imaginary on another host

### DISCUSSION THREAD

**Discussion by CoolnsX** [nextcloud/all-in-one#3629] [Questions] [answered]
Hey, thanks for this wonderful AIO, It helped me learning about reverse proxies and how to optimize it so that it can open faster

Right now I am using it in this way
- main nextcloud AIO on server with only imaginary as service available
- collabora office on another server and nginx proxy in another server (to help ease the load with main server, as main server also running jellyfin) 

everything works fine, so all I want is to shift nextcloud imaginary to another server and for that I need docker compose command --
even dockerfile commands are fine, I will compile it myself

**Comment by CoolnsX** [comment]
I really have concerns now --

~~- why nextcloud always resetting my configuration?~~
~~- why is it removing my "nextcloud office" app, everytime I restart the container?~~
- why is it not using the custom imaginary url that I provided?

PS :- imaginary and collabora is running on another host same as on "nginx proxy manager" server

My goal is to offload nextcloud services that works through http/https with nextcloud on another server

my server stack --
![image](https://github.com/nextcloud/all-in-one/assets/76195824/a617ab82-f52a-4cf5-aa8f-357c04095e95)

**Reply by Zoey2936** [reply, marked as answer]
>     * why is it removing my "nextcloud office" app, everytime I restart the container?

you can try to set NEXTCLOUD_KEEP_DISABLED_APPS=true

**Reply by CoolnsX** [reply]
Thanks !!!, 2 problems are fixed, but the reason I opened this discussion is not fulfilled yet, though I managed to make the image and container working but nextcloud is still not using "Imaginary", even if I set the preview path manually.

**Comment by CoolnsX** [comment]
It's working!!! POG

Thank you to the person that actually read and commented on this discussion.

**Comment by CoolnsX** [comment]
sorry I am re-opening it again, I only want to run my imaginary on another host.

**Comment by CoolnsX** [comment]
the thing is it was working fine, until I updated nextcloud and re-added the preview_imaginary_url with ip and port of host running it, but this time it's not working. like nextcloud not even checking if it's there, I tried to force preview generation so that I can check if nextcloud calling imaginary by inspecting imaginary logs through  "preview generator" app custom command "generate-all" but still no luck.
